packages/admin/src/table/filterbar/FilterBarPopoverFilter.tsx

```diff
@@ -0,0 +1,199 @@
+            cursor: "pointer",
+            display: "flex",
+
+            "&:after": {
+                borderTop: `4px solid ${theme.palette.grey[300]}`,
+                borderRight: "4px solid transparent",
+                borderLeft: "4px solid transparent",
+                position: "absolute",
+                display: "block",
+                right: "10px",
+                content: "''",
+                top: "50%",
+                height: 0,
+                width: 0,
+            },
+        },
+        labelWrapper: {
+            marginRight: "15px",
+            boxSizing: "border-box",
+        },
+        popoverContentContainer: {
+            minWidth: 300,
+            "& [class*='CometAdminFormFieldContainer-root']": {
+                boxSizing: "border-box",
+                padding: "20px",
+                marginBottom: 0,
+            },
+        },
+        buttonsContainer: {
+            borderTop: `1px solid ${theme.palette.grey[300]}`,
+            justifyContent: "space-between",
+            padding: "10px 15px",
+            display: "flex",
+        },
+        paper: {
+            border: "1px solid grey",
+            marginLeft: -1, //due to border of popover, but now overrideable with styling if needed
+        },
+    }),
+    { name: "CometAdminFilterBarPopoverFilter" },
+);
+
+export interface FilterBarPopoverFilterProps {
+    label: string;
+    dirtyFieldsBadge?: React.ComponentType<FilterBarActiveFilterBadgeProps>;
+    calcNumberDirtyFields?: (values: Record<string, any>) => number;
+}
+
+export function FilterBarPopoverFilter({
+    children,
+    label,
+    dirtyFieldsBadge,
+    calcNumberDirtyFields,
+}: React.PropsWithChildren<FilterBarPopoverFilterProps>) {
+    const FilterBarActiveFilterBadgeComponent = dirtyFieldsBadge ? dirtyFieldsBadge : FilterBarActiveFilterBadge;
+    const outerForm = useForm();
+    const [anchorEl, setAnchorEl] = React.useState<HTMLDivElement | null>(null);
+    const open = Boolean(anchorEl);
+
+    const handleClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+        setAnchorEl(event.currentTarget);
+    };
+
+    const { props: themeProps } = useTheme<ThemeOptions>();
+    const submitButtonProps =
+        themeProps && themeProps["CometAdminFilterBarPopoverFilter"] ? { ...themeProps["CometAdminFilterBarPopoverFilter"]?.submitButton } : {};
+    const resetButtonProps =
+        themeProps && themeProps["CometAdminFilterBarPopoverFilter"] ? { ...themeProps["CometAdminFilterBarPopoverFilter"]?.resetButton } : {};
+
+    const classes = useStyles();
+    const intl = useIntl();
+
+    return (
+        <div className={classes.root}>
+            <Form
+                onSubmit={(values) => {
+                    for (const name in values) {
+                        outerForm.change(name, values[name]);
+                    }
+                }}
+                initialValues={outerForm.getState().initialValues}
```

Comment:
you must use .values, not initialValues to support changing values from outside.

Possibly this would be broken: https://comet-admin.netlify.app/?path=/story/comet-admin-stack--stack-table-form-query-in-table (I'm not completely sure tough - needs to be tested)

Reply:
if i use values as my initialValues in my inner Form, what will be the value i can reset to ? because i will loose my initialValues if i use the outerForm values as my innerForm initialValues :)

Reply:
> what will be the value i can reset to

good question.

For the use-case for initialValues you have for POT (and that's the only use case we have so far that as initialValues at all), a reset should not use initialValues, undefined would be more appropriate (I guess). And that would be trivially to implement.

Reply:
any thoughts on that?

Reply:
i implemented it now as you said :) i think its just a UX thing, both ways are fine for me (really my honest opinion) :)